### ISSUE TITLE
Feature request: Provide right-to-left Arabic interface

### ISSUE THREAD

**Issue by kjantin** [huridocs/uwazi#933]
Allow Uwazi users to translate the interface into Arabic and have it read right-to-left.

Related to issue #789

**Comment by kjantin** [comment]
Our research:

TL;DR: It's not as complex as we initially thought and we may be able to implement this feature in way less time than we expected (1-3 weeks/1 person). Still, there are risks to consider.

Here is the good, the bad and the ugly.

Good:
- The samples you are seeing are the result of an automatic RTL css transformation library. I think they are pretty good, with some visible flaws here and there.
- This probably implies having to clean up a little bit the CSS for both LTR and RTL, which I consider desirable in terms of clarity, maintenance of code and UI responsiveness.
- Since our CSS are good quality in general, respecting responsiveness and standards, the RTL plug-in is achieving decent results by default.

Bad:
- There are several pitfalls related to how to handle numbers and dates in Arabic, since they are still consumed in LTR. This is one of the main risks.
- This will always add an overhead to developments involving UI.
- We currently don't have a dedicated person to UI, meaning all devs are going to have to wrestle with RTL. This factor could slow down development in general.

Ugly:
- This automation is never going to be perfect and manual tweaks will be always needed. So expect stuff not being displayed properly in the RTL version to be recurrent reported issue.
- Along with our base CSS, we use 3rd party CSS as well for maps, calendars, graphs. I haven't researched how well these libraries play with RTL.
- As always, evil is in the detail and it won't be until we start using RTL along with Arabic that we will realize how difficult it is.

**Comment by RafaPolit** [comment]
GENERAL ISSUES
- Progress bar: should it go from right to left on the top menu?
- Potential issue with brackets and parenthesis appearing at crazy points, specially when describing code like the settings URL links, etc.  Also on languages.
- How and who translates the interface (specially on settings?)

LIBRARY
- Beneath the search bar, the amount of documents and the sorting creteria is not correctly reverted
- The VIEW button in each card: should it point to the left instead of to the right and with the arrow on the right of the word?
- Load more - está funcionando?

SETTINGS
- Should 'return' arrows point to the right or to the left
- Pages - Check the resulting pages if they are correct

DOCUMENTS
- The boxing / rendering of the document (for selection) should NOT follow the language of the interface, but the language of the document?  What happens with mixed documents?
- When searching within document: check position of the number of page on results

**Comment by RafaPolit** [comment]
The Document now adds an LTR or RTL class according to the FRANC language of the file uploaded.  It defaults to LTR.

There appears to be an issue though, and that is: PDFJS appears to ALWAYS need the LTR direction in order to render the PDF correctly.  Even those that are created with RTL text.  This needs to be double checked with someone that has domain of a RTL language.

**Comment by RafaPolit** [comment]
The geolocation property needs to be addressed.  Clicking on the map works correctly and assigns the correct geolocation value, but the location marker is incorrectly positioned:
![screenshot from 2019-02-26 16-03-34](https://user-images.githubusercontent.com/4309019/53446169-500d0980-39e0-11e9-9ee3-f59b48013d3c.png)

**Comment by txau** [comment]
@RafaPolit maybe PDFjs 2 is supporting RTL?

**Comment by RafaPolit** [comment]
There is no RTL support, but we have solved the issue.  Briefly explained:
- PDF rendering is always done LTR, even for RTL PDFs.  This ensures that the positioning of text is always correct and that text layer is also correctly positioned.
- Overlay Text Layers are rendered in RTL which makes for a much better text selection experience.

With this, there is no offset from the rendered PDF and the text layers.  Also, the PDF looks as expected  and original positions are preserved (with LTR, word separation really got screwed).

**Comment by RafaPolit** [comment]
Bulk of the RTL development is ready.  There may be some minor tweaks to do further down the road, but overall functionality is done and should go into actual client testing mode soon.

**Comment by RafaPolit** [comment]
I have moved the 'nice to have's at this point to: #2220
